src/ui/server/controllers/insurance/policy-and-export/name-on-policy/index.test.ts

```diff
@@ -0,0 +1,168 @@
+import constructPayload from '../../../../helpers/construct-payload';
+import generateValidationErrors from './validation';
+import mapAndSave from '../map-and-save';
+import { Request, Response } from '../../../../../types';
+import { mockReq, mockRes, mockApplication } from '../../../../test-mocks';
+
+const {
+  INSURANCE: {
+    INSURANCE_ROOT,
+    POLICY_AND_EXPORTS: { NAME_ON_POLICY_SAVE_AND_BACK, CHECK_YOUR_ANSWERS },
+    PROBLEM_WITH_SERVICE,
+  },
+} = ROUTES;
+
+const {
+  NAME_ON_POLICY: { NAME, POSITION, SAME_NAME },
+} = POLICY_AND_EXPORTS_FIELD_IDS;
+
+describe('controllers/insurance/policy-and-export/name-on-policy', () => {
+  let req: Request;
+  let res: Response;
+  let refNumber: number;
+
+  jest.mock('../map-and-save');
+
+  mapAndSave.policyAndExport = jest.fn(() => Promise.resolve(true));
+
+  beforeEach(() => {
+    req = mockReq();
+    res = mockRes();
+
+    res.locals.application = mockApplication;
+    req.params.referenceNumber = String(mockApplication.referenceNumber);
+    refNumber = Number(mockApplication.referenceNumber);
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('pageVariables', () => {
+    it('should have correct properties', () => {
+      const result = pageVariables(refNumber);
+
+      const expected = {
+        FIELDS: {
+          NAME_ON_POLICY: {
+            ID: NAME,
+            ...FIELDS.NAME_ON_POLICY,
+          },
+          POSITION: {
+            ID: POSITION,
+            ...FIELDS.NAME_ON_POLICY[POSITION],
+          },
+        },
+        SAVE_AND_BACK_URL: `${INSURANCE_ROOT}/${mockApplication.referenceNumber}${NAME_ON_POLICY_SAVE_AND_BACK}`,
+      };
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('TEMPLATE', () => {
+    it('should have the correct template defined', () => {
+      expect(TEMPLATE).toEqual(TEMPLATES.INSURANCE.POLICY_AND_EXPORTS.NAME_ON_POLICY);
+    });
+  });
+
+  describe('FIELD_IDS', () => {
+    it('should have the correct FIELD_IDS', () => {
+      const expected = [NAME, POSITION];
+
+      expect(FIELD_IDS).toEqual(expected);
+    });
+  });
+
+  describe('get', () => {
+    it('should render template', async () => {
+      await get(req, res);
+
+      const expectedVariables = {
+        ...insuranceCorePageVariables({
+          PAGE_CONTENT_STRINGS: PAGES.INSURANCE.POLICY_AND_EXPORTS.NAME_ON_POLICY,
+          BACK_LINK: req.headers.referer,
+        }),
+        ...pageVariables(refNumber),
+        userName: getUserNameFromSession(req.session.user),
+        application: mockApplication,
+      };
+
+      expect(res.render).toHaveBeenCalledWith(TEMPLATE, expectedVariables);
+    });
+
+    describe('when there is no application', () => {
+      beforeEach(() => {
+        res.locals = mockRes().locals;
+      });
+
+      it(`should redirect to ${PROBLEM_WITH_SERVICE}`, async () => {
+        await get(req, res);
```

Comment:
No need for `await get` in any of the tests, the controller still has `async` as well, we don't need it - jest probably throwing a warning